docker/registry/storage/driver/tarfs/driver.go

```diff
@@ -0,0 +1,353 @@
+// Copyright 2021 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tarfs
```

Comment:
I don't think I understand what this is all doing and how much of it is borrowed code from existing implementations , would appreciate a quick walkthrough of the changes/

Reply:
Added a package level doc, but didn't want to document individual methods as this is just implementing the [driver](https://github.com/distribution/distribution/blob/main/registry/storage/driver/storagedriver.go#L56) interface.